### PR TITLE
[BEAM-1086] Upgrade to latest Gearpump snapshot

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by manuzhang** [comment]
R: @kennknowles

File: runners/gearpump/src/main/java/org/apache/beam/runners/gearpump/GearpumpRunner.java [hunk 1]
```diff
@@ -75,9 +74,10 @@ public static GearpumpRunner fromOptions(PipelineOptions options) {
 
   public <OutputT extends POutput, InputT extends PInput> OutputT apply(
       PTransform<InputT, OutputT> transform, InputT input) {
-    if (Window.Bound.class.equals(transform.getClass())) {
+    if (Window.Bound.class.equals(transform.getClass())
+        && isNullOrIdentityWindowFn(((Window.Bound) transform).getWindowFn())) {
```
Comment (kennknowles) [inline runners/gearpump/src/main/java/org/apache/beam/runners/gearpump/GearpumpRunner.java]: Does this change relate to the Gearpump upgrade? I feel like it is a separate concern. It is OK to bundle them in one PR, but would you mind splitting the commits?

```
git reset HEAD~
git add -p
# Choose just the Gearpump version changes
git commit -m 'Update Gearpump version'
git add -p
# Choose just the commits where you added the translator
git commit -m 'Add Window.Bound translator'
git add -p
# Choose just the other little changes
git commit -m 'Skip window assignment when windows don't change'
```

etc. It is up to you how many commits, but I would just like the Gearpump version bump to be isolated.

Comment (manuzhang) [inline runners/gearpump/src/main/java/org/apache/beam/runners/gearpump/GearpumpRunner.java]: Definitely. Thanks for the nice guide and I've split the commits a bit further.

**Comment by kennknowles** [comment]
LGTM, I'll merge it.

**Pull request merged**